### PR TITLE
expression: implement vectorized evaluation for `builtinSignSig`

### PR DESCRIPTION
**Pull request by gtygo**
### What problem does this PR solve? 
Implement vectorized evaluation for `builtinSignSig`, as listed in #12103 .

### What is changed and how it works?
Here is the benchmark, almost 7.3x faster than before:
```
BenchmarkVectorizedBuiltinMathFunc/builtinSignSig-VecBuiltinFunc-12                  	  262920	      4431 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinMathFunc/builtinSignSig-NonVecBuiltinFunc-12               	   36554	     32638 ns/op	       0 B/op	       0 allocs/op
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_math_vec_test.go [hunk 1]
```diff
@@ -23,7 +23,10 @@ import (
 )
 
 var vecBuiltinMathCases = map[string][]vecExprBenchCase{
-	ast.Sign: {},
+	ast.Sign: {
+		{retEvalType: types.ETInt, childrenTypes: []types.EvalType{types.ETInt}},
+		{retEvalType: types.ETInt, childrenTypes: []types.EvalType{types.ETInt}, geners: []dataGenerator{nil, nil}},
```
Comment (qw4990) [inline expression/builtin_math_vec_test.go]: It seems we don't need this case, which is duplicated with the one above it.

Comment (gtygo) [reply expression/builtin_math_vec_test.go]: That's a mistake. It's solved.

File: expression/builtin_math_vec.go [hunk 2]
```diff
@@ -807,11 +807,27 @@ func (b *builtinFloorIntToDecSig) vecEvalDecimal(input *chunk.Chunk, result *chu
...
+	if err := b.args[0].VecEvalReal(b.ctx, input, result); err != nil {
+		return err
+	}
+	i64s := result.Int64s()
+	for i := 0; i < len(i64s); i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		if i64s[i] == 0 {
+			result.SetNull(i, true)
```
Comment (qw4990) [inline expression/builtin_math_vec.go]: We shouldn't set it to `null` according to the row-based implementation.

Comment (gtygo) [reply expression/builtin_math_vec.go]: done, delete this line ,when i64s[i]==0, do nothing.

File: expression/builtin_math_vec_test.go [hunk 3]
```diff
@@ -23,7 +23,9 @@ import (
 )
 
 var vecBuiltinMathCases = map[string][]vecExprBenchCase{
-	ast.Sign: {},
+	ast.Sign: {
+		{retEvalType: types.ETInt, childrenTypes: []types.EvalType{types.ETInt}},
```
Comment (qw4990) [inline expression/builtin_math_vec_test.go]: The type of its child should be `ETReal`.

Comment (gtygo) [reply expression/builtin_math_vec_test.go]: done

File: expression/builtin_math_vec.go [hunk 4]
```diff
@@ -807,11 +807,25 @@ func (b *builtinFloorIntToDecSig) vecEvalDecimal(input *chunk.Chunk, result *chu
...
-	return false
+	return true
 }
 
 func (b *builtinSignSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	if err := b.args[0].VecEvalReal(b.ctx, input, result); err != nil {
+		return err
+	}
+	i64s := result.Int64s()
```
Comment (qw4990) [inline expression/builtin_math_vec.go]: We can't call `result.Int64s` since `result` is used to store real datums in `args[0].VecEvalReal`.

Comment (gtygo) [reply expression/builtin_math_vec.go]: oh!  i understand !  
I need to use `bufAllocator`  to store intermediate results.

File: expression/builtin_math_vec.go [hunk 5]
```diff
@@ -807,11 +807,34 @@ func (b *builtinFloorIntToDecSig) vecEvalDecimal(input *chunk.Chunk, result *chu
...
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETReal, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalReal(b.ctx, input, buf); err != nil {
+		return err
+	}
+	args := buf.Int64s()
```
Comment (Reminiscent) [inline expression/builtin_math_vec.go]: Here should be ```args := buf.Float64s()```

File: expression/builtin_math_vec.go [hunk 6]
```diff
@@ -807,11 +807,34 @@ func (b *builtinFloorIntToDecSig) vecEvalDecimal(input *chunk.Chunk, result *chu
...
+	i64s := result.Int64s()
+	for i := 0; i < len(i64s); i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		if args[i] > 0 {
+			i64s[i] = 1
+		} else if args[i] < 0 {
+			i64s[i] = -1
+		}
```
Comment (Reminiscent) [inline expression/builtin_math_vec.go]: Maybe you miss the situation that ``` args[i] == 0```

**Pull request merged**